src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -570,14 +570,10 @@ static int GetHostByNameHelper(const uint8_t* hostname, hostent** entry)
         hostent* result = reinterpret_cast<hostent*>(buffer);
         char* scratch = reinterpret_cast<char*>(&buffer[sizeof(hostent)]);
 
-        int getHostErrno;
+        int getHostErrno = 0;
         int err = gethostbyname_r(reinterpret_cast<const char*>(hostname), result, scratch, scratchLen, entry, &getHostErrno);
         switch (err)
```

Comment:
Is it correct to be switching on `err` rather than `getHostErrno`? (I'm actually asking, as I'm now unclear of the difference between the return value and the last arg.)  Is it possible `err` could be `0` but `getHostErrno` could be `ERANGE`?